go/arrow/array/list.go

```diff
@@ -618,19 +631,661 @@ func (b *baseListBuilder) UnmarshalJSON(data []byte) error {
+func NewListViewBuilderWithField(mem memory.Allocator, field arrow.Field) *ListViewBuilder {
+	offsetBldr := NewInt32Builder(mem)
+	sizeBldr := NewInt32Builder(mem)
+	return &ListViewBuilder{
+		baseListViewBuilder{
+			builder:         builder{refCount: 1, mem: mem},
+			values:          NewBuilder(mem, field.Type),
+			offsets:         offsetBldr,
+			sizes:           sizeBldr,
+			dt:              arrow.ListViewOfField(field),
+			appendOffsetVal: func(o int) { offsetBldr.Append(int32(o)) },
+			appendSizeVal:   func(s int) { sizeBldr.Append(int32(s)) },
+		},
+	}
+}
+
+func (b *baseListViewBuilder) Type() arrow.DataType {
+	switch dt := b.dt.(type) {
+	case *arrow.ListViewType:
+		f := dt.ElemField()
+		f.Type = b.values.Type()
+		return arrow.ListViewOfField(f)
+	case *arrow.LargeListViewType:
+		f := dt.ElemField()
+		f.Type = b.values.Type()
+		return arrow.LargeListViewOfField(f)
+	}
+	return nil
+}
+
+// NewLargeListViewBuilder returns a builder, using the provided memory allocator.
+// The created list-view builder will create a list whose elements will be of type etype.
+func NewLargeListViewBuilder(mem memory.Allocator, etype arrow.DataType) *LargeListViewBuilder {
+	offsetBldr := NewInt64Builder(mem)
+	sizeBldr := NewInt64Builder(mem)
+	return &LargeListViewBuilder{
+		baseListViewBuilder{
+			builder:         builder{refCount: 1, mem: mem},
+			values:          NewBuilder(mem, etype),
+			offsets:         offsetBldr,
+			sizes:           sizeBldr,
+			dt:              arrow.LargeListViewOf(etype),
+			appendOffsetVal: func(o int) { offsetBldr.Append(int64(o)) },
+			appendSizeVal:   func(s int) { sizeBldr.Append(int64(s)) },
+		},
+	}
+}
+
+// NewLargeListViewBuilderWithField takes a field rather than just an element type
+// to allow for more customization of the final type of the LargeListView Array
+func NewLargeListViewBuilderWithField(mem memory.Allocator, field arrow.Field) *LargeListViewBuilder {
+	offsetBldr := NewInt64Builder(mem)
+	sizeBldr := NewInt64Builder(mem)
+	return &LargeListViewBuilder{
+		baseListViewBuilder{
+			builder:         builder{refCount: 1, mem: mem},
+			values:          NewBuilder(mem, field.Type),
+			offsets:         offsetBldr,
+			sizes:           sizeBldr,
+			dt:              arrow.LargeListViewOfField(field),
+			appendOffsetVal: func(o int) { offsetBldr.Append(int64(o)) },
+			appendSizeVal:   func(o int) { sizeBldr.Append(int64(o)) },
+		},
+	}
+}
+
+// Release decreases the reference count by 1.
+// When the reference count goes to zero, the memory is freed.
+func (b *baseListViewBuilder) Release() {
+	debug.Assert(atomic.LoadInt64(&b.refCount) > 0, "too many releases")
+
+	if atomic.AddInt64(&b.refCount, -1) == 0 {
+		if b.nullBitmap != nil {
+			b.nullBitmap.Release()
+			b.nullBitmap = nil
+		}
+		b.values.Release()
+		b.offsets.Release()
+		b.sizes.Release()
+	}
+}
+
+func (b *baseListViewBuilder) appendDimensions(offset int, list_size int) {
```

Comment:
`list_size` should be `listSize` for go convensions

Reply:
Fixing :sweat_smile: